stdlib/public/core/Result.swift

```diff
@@ -76,8 +117,24 @@ public enum Result<Success, Failure: Error> {
   ///   instance.
   /// - Returns: A `Result` instance with the result of evaluating `transform`
   ///   as the new failure value if this instance represents a failure.
-  @inlinable
-  public func mapError<NewFailure>(
+  @_alwaysEmitIntoClient
+  public consuming func mapError<NewFailure>(
+    _ transform: (Failure) -> NewFailure
+  ) -> Result<Success, NewFailure> {
+    switch consume self {
+    case let .success(success):
+      return .success(consume success)
+    case let .failure(failure):
+      return .failure(transform(failure))
+    }
+  }
+}
+
+extension Result {
+  // TODO: Merge this back into the noncopyable variant once we have @_preInverseGenerics
```

Comment:
Are we going to merge this one back? Or outdated comment?

Reply:
Good catch, merged